### PR TITLE
privilege, executor: support `SET DEFAULT ROLE`

### PR DESCRIPTION
**Pull request by imtbkcat**
### What problem does this PR solve? 
Support `SET DEFAULT ROLE` gramma.
See https://dev.mysql.com/doc/refman/8.0/en/set-default-role.html for detail

### What is changed and how it works?
When `SET DEFAULT ROLE` for some users, we clear their data in `mysql.default_roles` and insert new default role information into `mysql.default_roles`.

Things need to be pay attention is handling error when insert transaction  failed.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

Side effects

 - Increased code complexity

### CODE REVIEW

**Comment by imtbkcat** [comment]
parser pr: https://github.com/pingcap/parser/pull/266

File: session/session.go [hunk 1]
```diff
@@ -1198,6 +1198,21 @@ func (s *session) Auth(user *auth.UserIdentity, authentication []byte, salt []by
 	user.AuthUsername, user.AuthHostname, success = pm.ConnectionVerification(user.Username, user.Hostname, authentication, salt)
 	if success {
 		s.sessionVars.User = user
+		// Load default active roles if login successed.
```
Comment (tiancaiamao) [inline session/session.go]: Could this be moved to somewhere like `session.go:loadCommonGlobalVariablesIfNeeded()` ?

Could the result be cached? Image that 1K connection connect to tidb simultaneously, tikv hot region will be a bottleneck for this operation.

Comment (imtbkcat) [reply session/session.go]: Ok, I will cache it.

**Comment by tiancaiamao** [comment]
LGTM

**Comment by tiancaiamao** [comment]
PTAL @jackysp @lysu

File: executor/simple.go [hunk 2]
```diff
@@ -87,11 +87,149 @@ func (e *SimpleExec) Next(ctx context.Context, req *chunk.RecordBatch) (err erro
...
 		err = e.executeRevokeRole(x)
+	case *ast.SetDefaultRoleStmt:
+		err = e.executeSetDefaultRole(x)
 	}
 	e.done = true
 	return err
 }
 
+func (e *SimpleExec) setDefaultRoleNone(s *ast.SetDefaultRoleStmt) error {
+	if _, err := e.ctx.(sqlexec.SQLExecutor).Execute(context.Background(), "begin"); err != nil {
```
Comment (jackysp) [inline executor/simple.go]: could we avoid repeatedly using the interface assert?

Comment (imtbkcat) [reply executor/simple.go]: PTAL @jackysp

**Pull request merged**